### ISSUE TITLE
Create and document interface to k_aug

### ISSUE THREAD

**Issue by adowling2** [Pyomo/pyomo#1485]
Currently, parmest can use k_aug (https://github.com/dthierry/k_aug) to calculate the reduced Hessian. Other projects such as IDAES may also use k_aug. I propose we design, create, and document a Pyomo interface to k_aug as part of the ``sensitivity_toolbox`` in ``Pyomo.contrib``.

What are all of the public projects that use k_aug? Here is a partial list:
- parmest
- KIPET (https://github.com/salvadorgarciamunoz/KIPET)
- Caprese

We can use these projects to guide the design of a k_aug interface. Specifically, what features are most frequently used? Besides these projects, I do not know of any examples/documentation of using k_aug with Pyomo.

@blnicho @Robbybp any thoughts?

**Comment by adowling2** [comment]
Here are some places where KIPET interfaces with k_aug:

https://github.com/salvadorgarciamunoz/kipet/blob/22b56bd1937265f983a4955ab9b8761f3230013f/kipet/library/MultipleExperimentsEstimator.py#L150

https://github.com/salvadorgarciamunoz/kipet/blob/22b56bd1937265f983a4955ab9b8761f3230013f/kipet/library/MultipleExperimentsEstimator.py#L1130

https://github.com/salvadorgarciamunoz/kipet/blob/22b56bd1937265f983a4955ab9b8761f3230013f/kipet/library/MultipleExperimentsEstimator.py#L1188

https://github.com/salvadorgarciamunoz/kipet/blob/22b56bd1937265f983a4955ab9b8761f3230013f/kipet/library/MultipleExperimentsEstimator.py#L1301

https://github.com/salvadorgarciamunoz/kipet/blob/22b56bd1937265f983a4955ab9b8761f3230013f/kipet/library/MultipleExperimentsEstimator.py#L1367

https://github.com/salvadorgarciamunoz/kipet/blob/22b56bd1937265f983a4955ab9b8761f3230013f/kipet/library/ParameterEstimator.py#L556

https://github.com/salvadorgarciamunoz/kipet/blob/22b56bd1937265f983a4955ab9b8761f3230013f/kipet/library/ParameterEstimator.py#L611

https://github.com/salvadorgarciamunoz/kipet/blob/22b56bd1937265f983a4955ab9b8761f3230013f/kipet/library/ParameterEstimator.py#L800

https://github.com/salvadorgarciamunoz/kipet/blob/22b56bd1937265f983a4955ab9b8761f3230013f/kipet/library/ParameterEstimator.py#L841

https://github.com/salvadorgarciamunoz/kipet/blob/22b56bd1937265f983a4955ab9b8761f3230013f/kipet/library/ParameterEstimator.py#L2085

TODO: How is this interface similar/different from the interface in parmest?

**Comment by dthierry** [comment]
What is parmest? Could you provide a link?

**Comment by blnicho** [comment]
@dthierry: https://pyomo.readthedocs.io/en/latest/contributed_packages/parmest/index.html

**Comment by Robbybp** [comment]
I believe all the code interfacing with k_aug is in this file https://github.com/Pyomo/pyomo/blob/master/pyomo/contrib/parmest/parmest.py

**Comment by adowling2** [comment]
@dthierry @Robbybp Yes, that is the only part of parmest that calls k_aug. I propose we create a documented interface (function) that calls k_aug to return a reduced Hessian. For example, PR #1475 uses the function `inv_reduced_hessian_barrier` from Pynumero.
